### ISSUE TITLE
Shutting down while partition merge is in progress results in panic

### ISSUE THREAD

**Issue by petr-tichy** [VictoriaMetrics/VictoriaMetrics#5944]
### Describe the bug

Restarting VM while partition merge is in progress results in panic.

### To Reproduce

```
curl localhost:8428/internal/force_merge
docker compose restart victoria-metrics
```

### Version

victoria-metrics-20240201-044014-heads-public-single-node-0-g8aaa828ba

### Logs

```
2024-03-09T10:08:00.954Z	info	VictoriaMetrics/lib/storage/table.go:243	starting forced merge for partition "2023_02"
2024-03-09T10:08:02.789Z	info	VictoriaMetrics/app/victoria-metrics/main.go:84	received signal terminated
2024-03-09T10:08:02.790Z	info	VictoriaMetrics/app/victoria-metrics/main.go:89	gracefully shutting down webservice at ":8428"
2024-03-09T10:08:02.797Z	info	VictoriaMetrics/app/victoria-metrics/main.go:94	successfully shut down the webservice in 0.008 seconds
2024-03-09T10:08:02.798Z	info	VictoriaMetrics/app/vmstorage/main.go:245	gracefully closing the storage at victoria-metrics-data
2024-03-09T10:08:02.799Z	panic	VictoriaMetrics/lib/storage/table.go:189	BUG: unexpected refCount=2 when closing the partition; probably there are pending searches
panic: BUG: unexpected refCount=2 when closing the partition; probably there are pending searches

goroutine 1 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage({0x7af209, 0x5}, {0x25a4480, 0x5a}, 0x4)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:301 +0xba4
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, {0x7af209, 0x5}, {0x7f2c06, 0x5b}, {0x2727e1c, 0x1, 0x1})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:141 +0x410
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:133
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:129
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*table).MustClose(0x2882cc0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/table.go:189 +0x184
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*Storage).MustClose(0x25c8000)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/storage.go:850 +0x80
github.com/VictoriaMetrics/VictoriaMetrics/app/vmstorage.Stop()
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmstorage/main.go:249 +0x114
main.main()
	github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-metrics/main.go:97 +0x964
```

### Used command-line flags

```
2024-03-09T10:08:03.860Z	info	VictoriaMetrics/lib/logger/flag.go:13	command-line flags
2024-03-09T10:08:03.860Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -inmemoryDataFlushInterval="1m0s"
2024-03-09T10:08:03.860Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -memory.allowedBytes="256MiB"
2024-03-09T10:08:03.860Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -retentionPeriod="10y"
2024-03-09T10:08:03.861Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -search.maxConcurrentRequests="3"
2024-03-09T10:08:03.861Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -search.maxMemoryPerQuery="128MiB"
2024-03-09T10:08:03.861Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -search.maxPointsSubqueryPerTimeseries="500000"
2024-03-09T10:08:03.861Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -search.maxQueryDuration="2m0s"
2024-03-09T10:08:03.861Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -search.maxQueueDuration="1m0s"
```

**Comment by hagen1778** [comment]
Hello @petr-tichy ! Looks like the mentioned version is built from custom commit, it is not an official release. Do you observe the same issue when running VM on official release [1.99](https://docs.victoriametrics.com/changelog/#v1990)?

**Comment by petr-tichy** [comment]
@hagen1778 I was running the [latest](https://hub.docker.com/layers/victoriametrics/victoria-metrics/latest/images/sha256-c1a68b2d86c1b557d3af5507ca626f987aafa52eed7c71d5024dc8070217230e?context=explore) docker image, but with [1.99](https://hub.docker.com/layers/victoriametrics/victoria-metrics/v1.99.0/images/sha256-2008aeebe2f4b849f90eba6693af10e0e614ae948aaf6c128a9bec527e61821b?context=explore) the result is the same.

**Comment by hagen1778** [comment]
> @hagen1778 I was running the [latest](https://hub.docker.com/layers/victoriametrics/victoria-metrics/latest/images/sha256-c1a68b2d86c1b557d3af5507ca626f987aafa52eed7c71d5024dc8070217230e?context=explore) docker image, but with [1.99](https://hub.docker.com/layers/victoriametrics/victoria-metrics/v1.99.0/images/sha256-2008aeebe2f4b849f90eba6693af10e0e614ae948aaf6c128a9bec527e61821b?context=explore) the result is the same. This is arm/v7 (RPI)

Thanks for confirming! Just in case, please avoid using :latest tag as it may contain experimental code. Use specific version or :stable instead.

**Comment by valyala** [comment]
The bugfix for this issue has been included in [v1.100.0 release](https://docs.victoriametrics.com/changelog/). Now VictoriaMetrics should gracefully stop [forced merge](https://docs.victoriametrics.com/#forced-merge) during graceful shutdown.

Closing the issue as fixed.
